multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1321,7 +1373,21 @@ def _top_k_average(
                 new_prefix="model",
             )
 
-        checkpoint = {"state_dict": avg_state_dict}
+        if not standalone:
+            checkpoint = {"state_dict": avg_state_dict}
+        else:
+            if strategy and hasattr(strategy, "strategy_name") and strategy.strategy_name == DEEPSPEED_STRATEGY:
+                checkpoint = {
+                    "state_dict": {
+                        name.partition("module.")[2]: param
+                        for name, param in strategy.model._zero3_consolidated_16bit_state_dict().items()
+                    }
+                }
+            else:
+                checkpoint = {
+                    "state_dict": {"model." + name: param for name, param in self._model.state_dict().items()}
+                }
```

Comment:
I guess this is the same as `checkpoint = {"state_dict": avg_state_dict}` in `if not standalone`, right?

Reply:
In the situation that not all parameters are updated, `avg_state_dict` does not contain all model weights, only the updated ones. If `standalone==True`, we want to save entire the model's weights, hence going through the model's entire `state_dict` in that case.